### PR TITLE
New unit tests for Delegation01, -02 and -03, and legacy tests are removed

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

This PR implements the scenarios and test zones for Delegation01, Delegation02 and Delegation03 defined in zonemaster/zonemaster#1305.

This PR is built on top of commit b276cf8e in PR #1393 to get updated version of `t/TestUtil.pm`.

Old legacy unit tests files are removed. The zones behind most of them will soon be removed.

In unit test file `t/Test-delegation.t` legacy tests and commented code have been removed. The file, however, cannot be rerecorded due to changes in zones not controlled by Zonemaster.

Some scenarios for Delegation01 cannot be included due to bug or bugs in the implementation (not in the scope of this PR):

```
t/Test-delegation01.t .. 1/? Untested scenarios:
	Scenario ENOUGH-DEL-NOT-CHILD cannot be tested.
	Scenario IPV4-AND-DEL-OK-NO-IPV6-CHILD cannot be tested.
	Scenario IPV6-AND-DEL-OK-NO-IPV4-CHILD cannot be tested.
	Scenario MISMATCH-DELEGATION-CHILD-1 cannot be tested.
	Scenario MISMATCH-DELEGATION-CHILD-2 cannot be tested.
```

Some scenarios for Delegation02 cannot be included due to bug or bugs in the implementation (not in scope of this PR):
```
t/Test-delegation02.t .. 1/? Untested scenarios:
	Scenario CHILD-NON-DISTINCT cannot be tested.
	Scenario DEL-NON-DISTINCT cannot be tested.
```

Please note that scenario CHILD-NON-DISTINCT and CHILD-NON-DISTINCT-UND are the same except for that the latter is undelegated. And the latter passes. The same for DEL-NON-DISTINCT and DEL-NON-DISTINCT-UND. This shows that the code does not handle normal and undelegated tests in the sama way.  The zone content "poisons" the delegation. 

No issues with Delegation03.

## Context

zonemaster/zonemaster#1305.

This PR is built on top of #1398. Commit e9060d17 does not belong to this PR.

## Changes

...

## How to test this PR

* The tests should pass. 
* Review zonemaster/zonemaster#1305.

### CODE REVIEW

File: t/Test-delegation01.t [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (marc-vanderwal) [inline t/Test-delegation01.t:235]: Some extraneous blank lines at the end of file:
```suggestion
```

File: t/Test-delegation02.t [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (marc-vanderwal) [inline t/Test-delegation02.t:186]: Some more extraneous blank lines:
```suggestion
```

**Review by marc-vanderwal — COMMENTED** [review]
Looks fine. Just some really minor nits.

**Comment by matsduf** [comment]
@marc-vanderwal, please re-review.

**Review by marc-vanderwal — APPROVED** [review]

**Comment by matsduf** [comment]
@marc-vanderwal, I do not know what "The merge-base changed after approval" really means. I did not do anything.

image

**Comment by marc-vanderwal** [comment]
Isn’t it because PRs are being merged in the `develop` branch and that this PR is being rebased continuously on the latest `develop`? I’m not sure either. Weird.

**Review by marc-vanderwal — APPROVED** [review]

**Comment by matsduf** [comment]
> Isn’t it because PRs are being merged in the `develop` branch and that this PR is being rebased continuously on the latest `develop`? I’m not sure either. Weird.

It has only been rebased once, and that was on top of ommit https://github.com/zonemaster/zonemaster-engine/commit/e9060d172362ad0db66b653a575cfaba86e6146d. I have never seen this behavior before.

**Comment by matsduf** [comment]
After the approval by @marc-vanderwal there as been no change, just execution of the Github bug.

**Pull request merged**